superset/assets/src/visualizations/deckgl/CategoricalDeckGLContainer.jsx

```diff
@@ -80,7 +80,7 @@ export default class CategoricalDeckGLContainer extends React.PureComponent {
   }
   getLayers(values) {
     const fd = this.props.slice.formData;
-    let data = [...this.props.data];
+    let data = [...this.props.payload.data.features];
```

Comment:
Here you're making a copy of `this.props.payload.data.features`, mutating it, and then assigning it back to `payload` in line 107. Here you can simply do:

```javascript
let data = this.props.payload.data.features;
```

And then as you mutate it, `payload.data.features` is also mutated. This way you can simply pass the original payload to `getLayer`.

Reply:
Did some test and found it does not work. I guess it is because the location that `data` references got changed so `payload.data.features` will not get updated.

This is part of the code:
```
    let data = [...this.props.payload.data.features];

    // Add colors from categories or fixed color
    data = this.addColor(data, fd);
```
If we change it to `let data = this.props.payload.data.features;` and then `data` get reassigned by `data = this.addColor(data, fd);`, it is not referencing to the location of `this.props.payload.data.features` anymore so `this.props.payload.data.features` will not be changed.